ckanext/iati_generator/auth/iati.py

```diff
@@ -124,3 +124,40 @@ def iati_file_list(context, data_dict):
         "success": False,
         "msg": toolkit._("Only sysadmins can list IATI files.")
     }
+
+
+def generate_organization_xml(context, data_dict):
+    """
+    Authorization for generating organization XML.
+    Only sysadmins or organization admins can generate XML for their organization.
+    """
+    owner_org = data_dict.get('owner_org')
+    if not owner_org:
+        return {
+            "success": False,
+            "msg": toolkit._("Missing owner_org parameter.")
+        }
```

Comment:
Use the organization for the package of the resource with the `FINAL_ORGANIZATION_FILE`.
We already have the `_allow_if_sysadmin_or_org_admin` function, we can re use it